core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -183,13 +184,21 @@ abstract ClientStream newSubstream(
 
   /** Adds grpc-previous-rpc-attempts in the headers of a retry/hedging RPC. */
   @VisibleForTesting
-  final Metadata updateHeaders(Metadata originalHeaders, int previousAttempts) {
+  final Metadata updateHeaders(
+      Metadata originalHeaders, int previousAttempts, int transparentRetryAttempts) {
     Metadata newHeaders = originalHeaders;
     if (previousAttempts > 0) {
       newHeaders = new Metadata();
       newHeaders.merge(originalHeaders);
       newHeaders.put(GRPC_PREVIOUS_RPC_ATTEMPTS, String.valueOf(previousAttempts));
     }
+    if (transparentRetryAttempts > 0) {
+      if (newHeaders == originalHeaders) {
+        newHeaders = new Metadata();
+        newHeaders.merge(originalHeaders);
+      }
+      newHeaders.put(GRPC_TRANSPARENT_RETRY_ATTEMPTS, String.valueOf(transparentRetryAttempts));
+    }
     return newHeaders;
```

Comment:
This must always be new headers. The transports can modify the metadata today (although I'm unsure if they actually do; it could be reasonable to institute a "no modify" policy for the stream).

Reply:
Done.